langfuse-core/src/media/LangfuseMedia.ts

```diff
@@ -194,6 +196,111 @@ class LangfuseMedia {
     };
   }
+
+  /**
+   * Replaces the media reference strings in an object with base64 data URIs for the media content.
+   *
+   * This method recursively traverses an object (up to a maximum depth of 10) looking for media reference strings
+   * in the format "@@@langfuseMedia:...@@@". When found, it fetches the actual media content using the provided
+   * Langfuse client and replaces the reference string with a base64 data URI.
+   *
+   * If fetching media content fails for a reference string, a warning is logged and the reference string is left unchanged.
+   *
+   * @param params - Configuration object
+   * @param params.obj - The object to process. Can be a primitive value, array, or nested object
+   * @param params.langfuseClient - Langfuse client instance used to fetch media content
+   * @param params.resolveWith - Optional. Default is "base64DataUri". The type of data to replace the media reference string with. Currently only "base64DataUri" is supported.
+   *
+   * @returns A deep copy of the input object with all media references replaced with base64 data URIs where possible
+   *
+   * @example
+   * ```typescript
+   * const obj = {
+   *   image: "@@@langfuseMedia:type=image/jpeg|id=123|source=bytes@@@",
+   *   nested: {
+   *     pdf: "@@@langfuseMedia:type=application/pdf|id=456|source=bytes@@@"
+   *   }
+   * };
+   *
+   * const result = await LangfuseMedia.resolveMediaReferences({
+   *   obj,
+   *   langfuseClient
+   * });
+   *
+   * // Result:
+   * // {
+   * //   image: "data:image/jpeg;base64,/9j/4AAQSkZJRg...",
+   * //   nested: {
+   * //     pdf: "data:application/pdf;base64,JVBERi0xLjcK..."
+   * //   }
+   * // }
+   * ```
+   */
+  public static async resolveMediaReferences<T>(params: {
+    obj: T;
+    langfuseClient: LangfuseCore;
+    resolveWith?: "base64DataUri";
+  }): Promise<T> {
+    const { obj, langfuseClient } = params;
+    const MAX_DEPTH = 10;
+
+    async function traverse<T>(obj: T, depth: number): Promise<T> {
+      if (depth > MAX_DEPTH) {
+        return obj;
+      }
+
+      // Handle string with potential media references
+      if (typeof obj === "string") {
+        const regex = /@@@langfuseMedia:.+@@@/g;
+        const referenceStringMatches = obj.match(regex);
+        if (!referenceStringMatches) {
+          return obj;
+        }
+
+        let result = obj;
+        const referenceStringToMediaContentMap = new Map<string, string>();
+
+        await Promise.all(
+          referenceStringMatches.map(async (referenceString) => {
+            try {
+              const parsedMediaReference = LangfuseMedia.parseReferenceString(referenceString);
+              const mediaData = await langfuseClient.getMediaById(parsedMediaReference.mediaId);
+              const mediaContent = await langfuseClient.fetch(mediaData.url, { method: "GET", headers: {} });
+              const base64MediaContent = Buffer.from(await mediaContent.arrayBuffer()).toString("base64");
```

Comment:
langfuse media token supports different representations (base64 is one of them). I think it would be more consistent if we either adapt to the type mentioned in the langfuse media token and/or if this can be overridden as an optional parameter

Reply:
Agree! This is why I added the `resolveWith` property in the params to control what string represenation should be used for the media content. I think this is a good start, and we can extend to more use cases on request. Do you think we should add it already in the first release?